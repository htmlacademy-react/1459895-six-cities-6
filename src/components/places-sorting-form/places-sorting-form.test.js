import React from 'react';
import {render} from '@testing-library/react';
import PlacesSortingForm from './places-sorting-form';
import * as redux from "react-redux";
import configureStore from "redux-mock-store";
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';

const mockStore = configureStore({});
describe(`Test PlacesSortingForm component`, () => {
  jest.spyOn(redux, `useSelector`);
  it(`PlacesSortingForm component should render correctly`, () => {
    const store = mockStore({
      MAIN: {option: `Popular`}
    });
    const history = createMemoryHistory();
    const {container} = render(
        <redux.Provider store={store}>
          <Router history={history}>
            <PlacesSortingForm />
          </Router>
        </redux.Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
