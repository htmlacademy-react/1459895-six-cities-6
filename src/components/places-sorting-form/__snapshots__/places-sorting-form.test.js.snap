// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test PlacesSortingForm component PlacesSortingForm component should render correctly 1`] = `
<div>
  <form
    action="#"
    class="places__sorting"
    method="get"
  >
    <span
      class="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      class="places__sorting-type"
      tabindex="0"
    >
      Popular
      <svg
        class="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          href="#icon-arrow-select"
          xlink="true"
        />
      </svg>
    </span>
  </form>
</div>
`;
